### PR TITLE
Allow templates in `initialize_project`

### LINKED ISSUE #264: Custom folder structure
**Is your feature request related to a problem? Please describe.**
I am starting to consider a custom folder structure for a class, and wonder if we can currently customize the structure created by `intiliaze_project`.

**Describe the solution you'd like**
I would like to have some sort of DrWatson.jl settings in `.julia/config/drwatson` that would describe the folder structure and the meaning of each folder, e.g. a "src" folder named "code" or a "notebooks" folder named "apps", etc.

**Describe alternatives you've considered**
I am considering not using DrWatson.jl's initiliaze_project functionality and only use the save/load functionality.

### PR DESCRIPTION
**Pull request by Datseris**
Closes #264

Okay, the idea is pretty simple. A template is just a vector of strings or pair=>vector{string}. Seems pretty clean to me, easy to adopt, easy to change. 

Any ideas/recommendations for a couple more pre-made templates are welcomed.

### CODE REVIEW

**Comment by Datseris** [comment]
cc @juliohm

**Comment by sebastianpech** [comment]
@Datseris quite busy at the moment, just skimmed your changes, will have a closer look tomorrow. One thing that came to my mind is, how are we going to treat our directory functions `datadir` etc? Now they aren't useful if one removes the corresponding directories from the template definition.

**Comment by Datseris** [comment]
> Now they aren't useful if one removes the corresponding directories from the template definition.

Yeah, the problem is, we can't, at least not without polluting global scope. It is possible to do some `@eval` hacks, but I don't agree with doing it. The user can use either `projectdir("new",.. )` or create their own versions as we instruct in the docs.

**Comment by sebastianpech** [comment]
I agree with you. I was thinking of creating eg. the `datadir` function only if the directory exists. We could do that during the project activation. Don't know if that's really necessary though.

File: src/project_setup.jl [hunk 1]
```diff
@@ -254,17 +200,31 @@ The new project remains activated for you to immidiately add packages.
...
-    force = false, readme = true, authors = nothing,
-    git = true, placeholder = false)
-
-    placeholder && !git && @warn "The placeholder files are created, but you need to "*
-    "manually commit them to your version control software OR initialize project with git = true"
+        force = false, readme = true, authors = nothing,
+        git = true, placeholder = false, template = DEFAULT_TEMPLATE
+    )
 
+    if git == false; placeholder = false; end
```
Comment (sebastianpech) [inline src/project_setup.jl:227]: What's this doing? Did you deliberately remove the warning for placeholder == true, git == false

**Review by sebastianpech — APPROVED** [review]
Besides the one comment, the code looks fine.

File: src/project_setup.jl (hunk 1 above)
Comment (Datseris) [reply src/project_setup.jl:227]: Yes. Well, as we say in the documentation string, `placeholder` has no reason to be `true` if `git` is `false`. So I just thought of making it `false` always if this is the case.

If this is the wrong way to go let me know and I'll revert.

Comment (sebastianpech) [reply src/project_setup.jl:227]: I'm good with that. The docstring still contains the info about the warning though.

**Review by sebastianpech — APPROVED** [review]

**Comment by Datseris** [comment]
Ok I'll polish, also add some info on the `datadir` business, and merge. If you have a template that you use all the time let me know.

**Pull request merged**